astropy/io/fits/util.py

```diff
@@ -583,10 +583,13 @@ def _array_from_file(infile, dtype, count):
         # their underlying file object, instead of the decompressed bytes
         read_size = np.dtype(dtype).itemsize * count
         s = infile.read(read_size)
-        array = np.frombuffer(s, dtype=dtype, count=count)
-        # copy is needed because np.frombuffer returns a read-only view of the
-        # underlying buffer
-        array = array.copy()
+        if s == b'':
+            array = np.ndarray(buffer=b'', dtype=np.dtype([]), shape=(0,))
```

Comment:
I would have hoped that you would not have to do an `if` statement at all. Did you try:
```
array = np.ndarray(buffer=s, dtype=dtype, shape=(count,))
```

Reply:
I was not sure while reading your comment ;), I will try!